### DISCUSSION TITLE
role already exists

### DISCUSSION THREAD

**Discussion by EetieDee** [serverless/serverless#12052] [General]
Hello, I am trying to deploy an application using  'serverless deploy', but I got the following message... What do I have to do, so that I can deploy?

```
× Stack x-dev failed to deploy
CREATE_FAILED: IamRoleLambdaExecution (AWS::IAM::Role)
x-dev-us-east-1-lambdaRole already exists
```

Hope someone can help. Thanks in advance.

ETD.

**Comment by Dexcover** [comment]
functions:
  myFunction:
    handler: handler.myFunction
    role: arn:aws:iam::123456789012:role/x-dev-us-east-1-lambdaRole

**Comment by edercarloscosta** [comment]
Hi @EetieDee

In this scenario I suggest few steps
 - First step execute `sls remove` for the clean up purpose, in the scenario when on your first deployment attempt fails some stuffs (dirty) remains there. 
 - Second step try to apply `sls deploy --verbose` and watch the logs in case something fails again you may to see an accurate log as possible.
 - Last one whether you keep seeing the same error message go manually to AWS Portal and remove this on IAM > Role section
 
I hope this may help you.
